### ISSUE TITLE
Gift Idea - Reminder Email

### ISSUE THREAD

**Issue by adrum** [monicahq/monica#157]
It would be nice to be able to set a `gift_reminder_date` that would send an email on the set date. 

I envision this being use when adding a Gift Idea. This is somewhat related to #154, but I think this could be it's own separate column, whereas the proposed `gift_date` field in #154 could just be a due date in the gift idea context. 

As in #155, I like to think about a gift idea, but sometimes hold off on purchasing the gift. It would be nice to be able to set a reminder date to be notified about a particular gift or gift idea. The reminder could serve whatever purpose, maybe the next time you want think about brainstorming the gift idea for someone's birthday or when you need to purchase it in order for to arrive in time by the due date (proposed in #154). 

If #155 is implemented, it would be great if this was an editable field where you could modify the reminder date, and it would continue to send a reminder on the this `gift_reminder_date` column, regardless if an email was previously sent about this particular date. 

Long story short, the email service would query for gift reminders by the `gift_reminder_date` column only, and not mark them as being sent already.
